components/brave_federated/data_stores/data_store_unittest.cc

```diff
@@ -35,86 +38,139 @@ namespace brave_federated {
+  ASSERT_TRUE(data_store_->Init(0, "test_federated_task", 50, 30));
+  ClearDatabase();
 }
 
 void DataStoreTest::TearDown() {
-  test_data_store_ = nullptr;
+  data_store_ = nullptr;
 }
 
-void DataStoreTest::ClearDB() {
-  EXPECT_TRUE(test_data_store_->DeleteLogs());
+void DataStoreTest::ClearDatabase() {
+  EXPECT_TRUE(data_store_->DeleteTrainingData());
 }
 
 size_t DataStoreTest::CountRecords() const {
-  sql::Statement s(test_data_store_->db_.GetUniqueStatement(
+  sql::Statement statement(data_store_->db_.GetUniqueStatement(
       "SELECT count(*) FROM test_federated_task"));
-  EXPECT_TRUE(s.Step());
-  return static_cast<size_t>(s.ColumnInt(0));
+  EXPECT_TRUE(statement.Step());
+  return static_cast<size_t>(statement.ColumnInt(0));
 }
 
-TestTaskLog DataStoreTest::TestTaskLogFromTestInfo(
-    const TestTaskLogTestInfo& info) {
-  return TestTaskLog(info.id, info.label,
-                     base::Time::Now() - base::Days(info.days_from_now));
+size_t DataStoreTest::CountTrainingInstances() const {
+  sql::Statement statement(data_store_->db_.GetUniqueStatement(
+      "SELECT count(DISTINCT training_instance_id) FROM test_federated_task"));
+  EXPECT_TRUE(statement.Step());
+  return static_cast<size_t>(statement.ColumnInt(0));
 }
 
-void DataStoreTest::AddAll() {
-  ClearDB();
-  for (size_t i = 0; i < std::size(test_task_log_test_db); ++i) {
-    TestTaskLog test_log = TestTaskLogFromTestInfo(test_task_log_test_db[i]);
-    test_data_store_->AddLog(test_log);
+DataStore::TrainingData DataStoreTest::TrainingDataFromTestInfo() {
+  DataStore::TrainingData training_data;
+
+  for (const auto& test_entry : kTestTrainingData) {
+    int training_instance_id = test_entry.training_instance_id;
+    mojom::CovariatePtr covariate = mojom::Covariate::New();
+    covariate->type = (mojom::CovariateType)test_entry.feature_name;
+    covariate->data_type = (mojom::DataType)test_entry.feature_type;
+    covariate->value = test_entry.feature_value;
+
+    training_data[training_instance_id].push_back(std::move(covariate));
+  }
+
+  return training_data;
+}
+
+void DataStoreTest::InitializeDataStore() {
+  ClearDatabase();
+  DataStore::TrainingData training_data = TrainingDataFromTestInfo();
+  for (auto& training_instance_pair : training_data) {
+    auto& training_instance = training_instance_pair.second;
+    AddTrainingInstance(std::move(training_instance));
   }
-  EXPECT_EQ(std::size(test_task_log_test_db), CountRecords());
+  EXPECT_EQ(std::size(kTestTrainingData), CountRecords());
+  EXPECT_EQ(std::size(training_data), CountTrainingInstances());
+}
+
+bool DataStoreTest::AddTrainingInstance(
+    std::vector<mojom::CovariatePtr> covariates) {
+  mojom::TrainingInstancePtr training_instance = mojom::TrainingInstance::New();
+  for (auto& covariate : covariates) {
+    training_instance->covariates.push_back(std::move(covariate));
+  }
+
+  return data_store_->AddTrainingInstance(std::move(training_instance));
 }
 
 // Actual tests
```

Comment:
not needed as should wrap the non member functions around an empty namespace

Reply:
Any non-member functions in implementation files should be wrapped with an empty namespace. Please check other instances